Gems/ROS2/Code/Source/Gripper/VacuumGripperComponent.cpp

```diff
@@ -0,0 +1,352 @@
+    }
+
+    void VacuumGripperComponent::ReleaseGrippedObject()
+    {
+        m_tryingToGrip = false;
+        if (m_vacuumJoint == AzPhysics::InvalidJointHandle)
+        {
+            return;
+        }
+        auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AzPhysics::SceneHandle defaultSceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+        AZ_Assert(defaultSceneHandle != AzPhysics::InvalidSceneHandle, "Invalid default physics scene handle");
+        // Wake up the body to prevent it from not moving after release
+        AzPhysics::RigidBody* grippedRigidBody = nullptr;
+        Physics::RigidBodyRequestBus::EventResult(grippedRigidBody, m_grippedObjectInEffector, &Physics::RigidBodyRequests::GetRigidBody);
+        if (grippedRigidBody)
+        {
+            grippedRigidBody->ForceAwake();
+        }
+        sceneInterface->RemoveJoint(defaultSceneHandle, m_vacuumJoint);
+        m_vacuumJoint = AzPhysics::InvalidJointHandle;
+    }
+
+    void VacuumGripperComponent::OnImGuiUpdate()
+    {
+        ImGui::Begin("VacuumGripperDebugger");
+
+        AZStd::string grippedObjectInEffectorName;
+        AZ::ComponentApplicationBus::BroadcastResult(
+            grippedObjectInEffectorName, &AZ::ComponentApplicationRequests::GetEntityName, m_grippedObjectInEffector);
+
+        ImGui::Text("Grippable object : %s", grippedObjectInEffectorName.c_str());
+        ImGui::Text("Vacuum joint created : %d ", m_vacuumJoint != AzPhysics::InvalidJointHandle);
+
+        ImGui::Checkbox("Gripping", &m_tryingToGrip);
+
+        if (ImGui::Button("Grip Command "))
+        {
+            m_tryingToGrip = true;
+        }
+
+        if (ImGui::Button("Release Command"))
+        {
+            ReleaseGrippedObject();
+        }
+        ImGui::End();
+    }
+
+    AZ::Outcome<void, AZStd::string> VacuumGripperComponent::GripperCommand(float position, float maxEffort)
+    {
+        AZ_Trace("VacuumGripper", "GripperCommand %f\n", position);
+        m_cancelGripperCommand = false;
+        if (position == 0.0f)
+        {
+            m_tryingToGrip = true;
+        }
+        else
+        {
+            ReleaseGrippedObject();
+        }
+        return AZ::Success();
+    }
+
+    AZ::Outcome<void, AZStd::string> VacuumGripperComponent::CancelGripperCommand()
+    {
+        ReleaseGrippedObject();
+        m_cancelGripperCommand = true;
+        return AZ::Success();
+    }
+
+    bool VacuumGripperComponent::HasGripperCommandBeenCancelled() const
+    {
+        if (m_cancelGripperCommand && m_vacuumJoint == AzPhysics::InvalidJointHandle)
+        {
+            return true;
+        }
+        return false;
+    }
+
+    float VacuumGripperComponent::GetGripperPosition() const
+    {
+        return m_vacuumJoint == AzPhysics::InvalidJointHandle ? 0.0f : 1.0f;
```

Comment:
I think we should return whether we are trying to grip, not successfully gripping. Currently, if we request a position 0 with no object to grip, it will return a success, but the gripper position from GetGripperPosition will remain unchanged. Moreover, I think the current way causes a delay (when i send a gripper command that successfully grips the object, it can first send the feedback and only then change the gripper position).
Also, I think the current condition should be negated.
```suggestion
        return m_tryingToGrip ? 0.0f : 1.0f;
```